src/python/pants/engine/process.py

```diff
@@ -45,10 +43,10 @@ def __init__(
         input_digest: Digest = EMPTY_DIGEST,
         working_directory: Optional[str] = None,
         env: Optional[Dict[str, str]] = None,
+        append_only_caches: Optional[Set[str]] = None,
```

Comment:
Better to take `Iterable` and call `set()` in the constructor. See https://pants.readme.io/docs/style-guide#prefer-protocols-duck-types-for-parameters.

Reply:
`Iterable` and set have pretty different semantics... the `set` type is intended to declare: 1) order doesn't matter, 2) dupes aren't allowed. If there is an "abstract" `Set` type of some sort, could do that?

Reply:
There's `typing.AbstractSet`, which will allow it to work with `FrozenOrderedSet` and `OrderedSet`.